src/benchmarktool/result/ods_gen.py

```diff
@@ -0,0 +1,422 @@
+            inst_sheet = odsfile.new_sheet("Instances")
+            for line in range(len(self.inst_sheet.content.index)):
+                inst_sheet.writerow(list(self.inst_sheet.content.iloc[line]))
+            class_sheet = odsfile.new_sheet("Classes")
+            for line in range(len(self.class_sheet.content.index)):
+                class_sheet.writerow(list(self.class_sheet.content.iloc[line]))
+
+
+# pylint: disable=too-many-instance-attributes
+class Sheet:
+    """
+    Class representing an ODS sheet.
+
+    (previously called Table/ResultTable)
+    """
+
+    def __init__(
+        self, benchmark: "result.BenchmarkMerge", measures: Any, name: str, ref_sheet: Optional["Sheet"] = None
+    ):
+        """
+        Initialize sheet.
+
+        Keyword arguments:
+        benchmark   - BenchmarkMerge object
+        measures    - Measures to be displayed
+        name        - Name of the sheet
+        refSheet    - Reference sheet
+        """
+        # dataframe resembling almost final ods form
+        self.content = pd.DataFrame()
+        # name of the sheet
+        self.name = name
+        # evaluated benchmarks
+        self.benchmark = benchmark
+        # dataframes containing result data, use these for calculations
+        self.system_blocks: dict[tuple[Any, Any], SystemBlock] = {}
+        # types of measures
+        self.types: dict[str, str] = {}
+        # measures to be displayed
+        self.measures = measures
+        # machines
+        self.machines: set["result.Machine"] = set()
+        # sheet for references
+        self.ref_sheet = ref_sheet
+        # references for summary generation
+        self.summary_refs: dict[str, Any] = {}
+
+        # first column
+        self.content[0] = None
+        # setup rows for instances/benchmark classes
+        if self.ref_sheet is None:
+            row = 2
+            for benchclass in benchmark:
+                for instance in benchclass:
+                    self.content.loc[row] = instance.benchclass.name + "/" + instance.name
+                    row += instance.maxRuns
+        else:
+            row = 2
+            for benchclass in benchmark:
+                self.content.loc[row] = benchclass.name
+                row += 1
+
+        self.result_offset = row
+        self.content.loc[self.result_offset + 1] = "SUM"
+        self.content.loc[self.result_offset + 2] = "AVG"
+        self.content.loc[self.result_offset + 3] = "DEV"
+        self.content.loc[self.result_offset + 4] = "DST"
+        self.content.loc[self.result_offset + 5] = "BEST"
+        self.content.loc[self.result_offset + 6] = "BETTER"
+        self.content.loc[self.result_offset + 7] = "WORSE"
+        self.content.loc[self.result_offset + 8] = "WORST"
+        # fill missing rows
+        self.content = self.content.reindex(list(range(self.content.index.max() + 1)))
+
+    # pylint: disable=too-many-nested-blocks, too-many-branches
```

Comment:
Maybe set globally or refactor because it seems like one could easily add some internal functions to handle results, cells, ...